UI: Consistent code for anchor elements, specifically external links

RachelElysia opened the pull request:
Cerra #4465

**Fixes**
- External link anchor elements attributes are consistently ordered: `className` `href` `target` `rel`
- Img elements attributes are consistently ordered: `className` `src` `alt` and use the same alt text
- External links all use the same icon `icon-external-link-12x12@2x.png` (except privacy policy link on Settings page)
- Replace `FleetIcon` link and delete duplicate icon in `/assets`
- All external link icons are separated by the link text using `&nbsp;`
- Anchor elements and anchor image elements styled globally

**Manual QA Screenshots of Anchor tags**
- DataError.tsx (card size and page size)
<img width="1458" alt="Screen Shot 2022-07-08 at 12 14 35 PM" src="https://user-images.githubusercontent.com/71795832/178030774-49f09636-2f6f-48c4-abe4-e1a67b366458.png">
<img width="917" alt="Screen Shot 2022-07-08 at 12 14 09 PM" src="https://user-images.githubusercontent.com/71795832/178030775-3bba32f0-1364-4d09-a0e8-f4e544bc9690.png">
- PackInfoSidePanel.jsx
<img width="318" alt="Screen Shot 2022-07-08 at 12 22 19 PM" src="https://user-images.githubusercontent.com/71795832/178031827-814576ec-7f1a-4a48-bb4d-d3d287a95fad.png">
- CoreLayout.tsx expiration message
<img width="896" alt="Screen Shot 2022-07-08 at 12 24 38 PM" src="https://user-images.githubusercontent.com/71795832/178032289-5b307195-e954-4d46-ab2e-62f629deba9f.png">
- ForgotPasswordPage.tsx
<img width="420" alt="Screen Shot 2022-07-08 at 12 31 20 PM" src="https://user-images.githubusercontent.com/71795832/178033188-8ea0e4f1-22af-4d22-8efb-3cbd0079d3c8.png">
- Homepage.tsx
<img width="564" alt="Screen Shot 2022-07-08 at 12 26 40 PM" src="https://user-images.githubusercontent.com/71795832/178032553-63e9610e-ef87-4513-81f2-aee15f6b5f43.png">
<img width="512" alt="Screen Shot 2022-07-08 at 12 26 35 PM" src="https://user-images.githubusercontent.com/71795832/178032554-ea159062-1517-4fcf-b31f-bdc51a484b3b.png">
- LearnFleet.tsx
- WelcomeHost/__style.scss
- UserSettingsPage.tsx
<img width="389" alt="Screen Shot 2022-07-08 at 12 37 59 PM" src="https://user-images.githubusercontent.com/71795832/178034172-7785249b-baa1-40dd-933f-7c69be3b4af2.png">
- Agents.tsx
<img width="620" alt="Screen Shot 2022-07-08 at 12 40 02 PM" src="https://user-images.githubusercontent.com/71795832/178034473-6d065768-2868-4067-8366-05526970d593.png">
- FleetDesktop.tsx
<img width="899" alt="Screen Shot 2022-07-08 at 12 43 25 PM" src="https://user-images.githubusercontent.com/71795832/178034981-98d6e8d1-1668-47b3-b5fd-d2808fa7c324.png">
- Statistics.tsx
<img width="664" alt="Screen Shot 2022-07-08 at 12 43 55 PM" src="https://user-images.githubusercontent.com/71795832/178035074-8cced776-d0a0-4470-802d-0005b89397a0.png">
- IntegrationsPage.tsx
- CreateIntegration.tsx
<img width="545" alt="Screen Shot 2022-07-08 at 2 58 17 PM" src="https://user-images.githubusercontent.com/71795832/178054385-2652b009-11f9-49d4-b673-51e373bb6ce3.png">
- AgentOptionsPage.tsx
<img width="686" alt="Screen Shot 2022-07-08 at 2 58 55 PM" src="https://user-images.githubusercontent.com/71795832/178054414-ce90ca90-cc9d-42cb-8525-b92bf2ea1450.png">
- TeamManagementPage.tsx
<img width="324" alt="Screen Shot 2022-07-08 at 3 06 25 PM" src="https://user-images.githubusercontent.com/71795832/178055240-b0fc9f1a-8794-49b7-994e-9c77a10a7fe7.png">
- UserForm.tsx
<img width="543" alt="Screen Shot 2022-07-08 at 3 04 40 PM" src="https://user-images.githubusercontent.com/71795832/178054960-c5eca037-307c-42db-80d2-7d765d7a8b43.png">
- DeviceUserPage/InfoModal.tsx
<img width="538" alt="Screen Shot 2022-07-08 at 2 59 16 PM" src="https://user-images.githubusercontent.com/71795832/178054460-e6d1cd78-e147-4c37-b82f-3337c8b94141.png">
- hosts/details/cards/EmptyState/EmptyState.tsx
- hosts/details/cards/Policies/Policies.tsx
<img width="849" alt="Screen Shot 2022-07-08 at 3 11 13 PM" src="https://user-images.githubusercontent.com/71795832/178055938-80d13a36-e71f-4efe-942f-59ed83094d2b.png">
- ManagePoliciesPage PreviewPayloadModal.tsx
<img width="548" alt="Screen Shot 2022-07-08 at 3 03 41 PM" src="https://user-images.githubusercontent.com/71795832/178054855-733155e5-1554-4867-b39a-eb0df97f12df.png">
- ManagePoliciesPage PreviewTicketModal.tsx
<img width="538" alt="Screen Shot 2022-07-08 at 3 03 32 PM" src="https://user-images.githubusercontent.com/71795832/178054867-45c60139-ec96-49a1-bf1c-5ff7bdcbd128.png">
- PolicyPage/PolicyPage.tsx (run live query error)
- QueriesListWrapper.tsx
<img width="471" alt="Screen Shot 2022-07-08 at 3 18 14 PM" src="https://user-images.githubusercontent.com/71795832/178056948-0e3d6042-f6f5-44f6-98ae-f01d6646c4e6.png">
- QueryPage.tsx (run live query error)
- ScheduleEditorModal.tsx
<img width="1030" alt="Screen Shot 2022-07-08 at 3 01 32 PM" src="https://user-images.githubusercontent.com/71795832/178054581-9a6ff209-03eb-41e8-8488-3abcfdec6d1e.png">
- ManageSoftwarePage.tsx
<img width="702" alt="Screen Shot 2022-07-08 at 3 16 36 PM" src="https://user-images.githubusercontent.com/71795832/178056592-18c24525-1e6b-4393-b577-3fc8001deee8.png">
- ManageSoftwarePage PreviewPayloadModal.tsx
<img width="1197" alt="Screen Shot 2022-07-08 at 2 59 46 PM" src="https://user-images.githubusercontent.com/71795832/178054329-5e6dcd6b-47d3-4948-8e67-5fe8fa888ad4.png">
- SoftwareDetailsPage VulnTableConfig.tsx
<img width="515" alt="Screen Shot 2022-07-08 at 3 02 25 PM" src="https://user-images.githubusercontent.com/71795832/178054691-4b60a8c9-af4a-4a2f-9b8e-16c63b2850d8.png">
- SoftwareDetailsPage Vulnerabilities.tsx
<img width="608" alt="Screen Shot 2022-07-08 at 3 14 35 PM" src="https://user-images.githubusercontent.com/71795832/178056282-dc31f987-2dbe-4d52-92d2-186318488b87.png">
- EmptySoftware.tsx



# Checklist for submitter

If some of the following don't apply, delete the relevant line.

- [x] Changes file added for user-visible changes (in `changes/` and/or `orbit/changes/`).
- [x] Manual QA for all new/changed functionality

lukeheath commented on frontend/pages/ForgotPasswordPage/ForgotPasswordPage.tsx:
```diff
@@ -59,13 +59,13 @@ const ForgotPasswordPage = () => {
               rel="noopener noreferrer"
             >
-              Password reset FAQ
+              Password reset FAQ&nbsp;
+              <img
+                className="icon-external"
+                src={ExternalLinkIcon}
+                alt="Open external link"
+              />
```
Instead of using `&nbsp;` to create the space, please create a global `icon-external` class that has `left-padding` or `left-margin` applied. That way, the space can be created once instead of every time it's implemented.

RachelElysia replied:
The first solution uses &nbsp;

The second solution uses adding span to the last word of every anchor link

I tried using margin-right: -15px, but that only messes up the following text after the anchor link.

lukeheath replied:
I pulled it down and it looks like even with &nbsp; it's still wrapping the image. Are you seeing something different? 

![image](https://user-images.githubusercontent.com/2495927/187547879-793efaec-a630-4776-ad4d-6e880a3f5098.png)

lukeheath reviewed (changes requested):
Thanks for tackling this! I know it was already part of the pattern, but as part of this effort let's remove the use of `&nbsp;` next to the links to create space. Instead, let's create a global class that can be applied to the external link icon to create the space, and clear out all the extraneous `&nbsp;`.

RachelElysia commented:
@lukeheath I removed all the &nbsp; and replaced the one's I saw wrapping with inline styling so there's less confusion why the span is around the last word + the image.

<img width="542" alt="Screen Shot 2022-09-02 at 1 18 34 PM" src="https://user-images.githubusercontent.com/71795832/188205665-fb4a717d-3a0e-48cf-b1b5-e28c5f5d9c26.png">

lukeheath commented:
> @lukeheath I removed all the `&nbsp;` and replaced the one's I saw wrapping with inline styling so there's less confusion why there's a span is around the last word + the image.

Not to nitpick, but I think a class called `no-wrap` would be equally clear, and may be helpful elsewhere. It's also hard for me to see the `style` attribute used 😬 

Would you please create a `.no-wrap` style and put it in `frontend/styles/global/_styles.scss`? Root specificity should be enough, but if not you can do `#app span.no-wrap` to get more.

lukeheath reviewed (approved):
This is awesome! Thanks for the great detail work on finding and patching this ✨
